src/SBKolmogorov.cpp

```diff
@@ -390,33 +385,42 @@ namespace galsim {
             dbg<<"f("<<r<<") = "<<val<<std::endl;
             _radial.addEntry(r,val);
 
-            // Accumulate int(r*f(r)) / dr  (i.e. don't include 2*pi*dr factor as part of sum)
-            // We are stepping by dlogr = dr/r, so accumulate r*r*f(r).
-            if (sum == 0.) {
-                // At small r, f(r) is very close to constant. So first sum = pi r^2 f(r).
-                sum = M_PI * r*r*val;
-            } else {
-                sum += r*r * val;
-            }
-            xdbg<<"sum = "<<sum<<"  thresh1 = "<<thresh1<<std::endl;
-            xdbg<<"sum*2*pi*dlogr "<<sum*2.*M_PI*dlogr<<std::endl;
-            if (R == 0. && sum > thresh1) R = r;
-            if (hlr == 0. && sum > thresh0) hlr = r;
-
-            // At high r, the profile is well approximated by a power law, F ~ r^-3.68
-            // The integral of the missing flux out to infinity is int_r^inf F(r) r dr = F r^2/1.68
-            xdbg<<"F r^2/1.68 = "<<val*r*r/1.68<<"  thresh2 = "<<thresh2<<std::endl;
-            if (hlr != 0 && val * r * r / 1.68 < thresh2) break;
+            // At high r, the profile is well approximated by a power law, F ~ r^-3.67
+            // The integral of the missing flux out to infinity is int_r^inf F(r) r dr = F r^2/1.67
+            xdbg<<"F r^2/1.67 = "<<val*r*r/1.67<<"  thresh = "<<thresh<<std::endl;
+            if (val * r * r / 1.67 < thresh) break;
         }
         _radial.finalize();
         dbg<<"Done loop to build radial function.\n";
+
+        // The large r behavior of F(r) is well approximated by a power law, F ~ r^-3.67
+        // This affords an easier calculation of R for stepk than numerically accumulating
+        // the integral.
+        // F(r) = F1 (r/r1)^-n
+        // int_r^inf F(r) 2pi r dr = folding_threshold
+        // 2pi F1 r1^n / (n-2) f_t = R^(n-2)
+        double r1 = _radial.argMax();
+        double F1 = _radial.lookup(r1);
+        double r2 = r1 * (1-dlogr);
+        double F2 = _radial.lookup(r2);
+        dbg<<"r1,F1 = "<<r1<<','<<F1<<std::endl;
+        dbg<<"r2,F2 = "<<r2<<','<<F2<<std::endl;
+        // power law index = dlog(F)/dlog(r)
+        double n = -(std::log(F2)-std::log(F1)) / (std::log(r2)-std::log(r1));
+        // For high folding_threshold, n can come out too low (e.g. < 2), which is bad.
+        // We know the right answer is close to 3.67, so just use that if n < 3.6.
+        if (n < 3.6) n = std::max(n, 3.67);
```

Comment:
Feels a little weird to introduce a discontinuity at n=3.6.  Can we just always apply `n = std::max(n, 3.67)`?

Reply:
Yeah, I wasn't quite sure what the best strategy here is.  Kolmogorov is a specific profile, so there is a right answer.  I originally had just hard-coded `n=3.67`.  But then I thought it might be better to let the code determine it, rather than introduce a special number.  That felt more principled to me, especially since as folding_threshold gets smaller, the estimate should get more accurate, so sticking to 3.67 might be introducing unwarranted inaccuracies in some (extreme) cases. 
But then it failed when folding_threshold was very high, like 0.05 or something, so I added this test, which I agree doesn't feel very principled anymore.
Maybe I'll try to get Maple or Wolfram to figure out a more precise number and just hard code it again.

Reply:
I couldn't coerce either Maple or Wolfram to give me the analytic asymptote.  Maple gave me a closed form solution for the Hankel transform -- a crazy combination of hypergeometric functions; see below for a -> ascii printout of the result.  But any kind of evaluation quickly became inaccurate as r increased, so it couldn't manage anything in the high r limit.

However, emprically, as I increased the accuracy of the calculations, I got answers that seemed to get quite close to 11/3 exactly.  Numbers like 3.66675 for instance.  So I'm putting forth a conjecture that it is exactly 11/3, since that seems like the kind of exponent that might emerge from Kolmogorov turbulence theory.  But I haven't proved it.

Also, the VonKarman gave basically the same answer even for rather small L0 values.  Even when L0=r0, I was still getting n=3.66628.  Once L0 << r0, it started to diverge, but I don't think we care about that limit.  So I just set n=11/3 always in that case too.

```
                          //                     
            1             ||         8          /
------------------------- ||1309 Pi r  hypergeom|
             /3\      /4\ \\                    \
5000000 GAMMA|-| GAMMA|-|                        
             \5/      \5/                        

  [9   16  37  47  26]  [   6  6  7  8  8  9  9]       729     10\ 
  [--, --, --, --, --], [1, -, -, -, -, -, -, -], - --------- r  | 
  [10  15  30  30  15]  [   5  5  5  5  5  5  5]    156250000    / 

       /4\    /1   \            6          /[7   13  31  41  23]  
  GAMMA|-| sin|- Pi| - 7040 Pi r  hypergeom|[--, --, --, --, --], 
       \5/    \5   /                       \[10  15  30  30  15]  

  [4  4     6  7  7  8  8]       729     10\      /3\    /2   \          
  [-, -, 1, -, -, -, -, -], - --------- r  | GAMMA|-| sin|- Pi| + 300000 
  [5  5     5  5  5  5  5]    156250000    /      \5/    \5   /          

   4          /[1  2  5  7  4]  [3  3  4  4  6  6  7  7]  
  r  hypergeom|[-, -, -, -, -], [-, -, -, -, -, -, -, -], 
              \[2  3  6  6  3]  [5  5  5  5  5  5  5  5]  

       729     10\      /3\      /4\    /2   \    /1   \           
  - --------- r  | GAMMA|-| GAMMA|-| sin|- Pi| sin|- Pi| - 1920000 
    156250000    /      \5/      \5/    \5   /    \5   /           

   2          /[3   7   19  29  17]  [2  2  3  3  4     6  6]  
  r  hypergeom|[--, --, --, --, --], [-, -, -, -, -, 1, -, -], 
              \[10  15  30  30  15]  [5  5  5  5  5     5  5]  

                           2                                     
       729     10\      /4\       /3\    /2   \    /1   \        
  - --------- r  | GAMMA|-|  GAMMA|-| sin|- Pi| sin|- Pi| + 9600\
    156250000    /      \5/       \5/    \5   /    \5   /        

               /[1   4   13  23  14]  [1  1  2  2  3  4  4   ]  
  000 hypergeom|[--, --, --, --, --], [-, -, -, -, -, -, -, 1], 
               \[10  15  30  30  15]  [5  5  5  5  5  5  5   ]  

                           2                             \       
       729     10\      /3\       /4\    /2   \    /1   \|    /2 
  - --------- r  | GAMMA|-|  GAMMA|-| sin|- Pi| sin|- Pi|| sin|- 
    156250000    /      \5/       \5/    \5   /    \5   //    \5 

               \
    \    /1   \|
  Pi| sin|- Pi||
    /    \5   //
```

Reply:
I bet 11/3 is right too.  Thanks for looking into this.